### ISSUE TITLE
Lot of warnings from within the macro

### ISSUE THREAD

**Issue by cecton** [paritytech/jsonrpsee#106]
I'm getting a lot of warnings during the compilation:

```
   |
   = note: `#[warn(unused_variables)]` on by default

warning: unused variable: `method`
  --> test/parachain/tests/integration_test.rs:50:2
   |
50 |     Chain {
   |     ^^^^^ help: consider prefixing with an underscore: `_method`

warning: unused variable: `request`
  --> test/parachain/tests/integration_test.rs:52:3
   |
52 |         fn current_block_hash() -> PHash;
   |         ^^ help: consider prefixing with an underscore: `_request`

warning: unused variable: `method`
  --> test/parachain/tests/integration_test.rs:61:2
   |
61 |     State {
   |     ^^^^^ help: consider prefixing with an underscore: `_method`

warning: unused variable: `request`
  --> test/parachain/tests/integration_test.rs:63:3
   |
63 |         fn runtime_version() -> RuntimeVersion;
   |         ^^ help: consider prefixing with an underscore: `_request`

warning: variant is never constructed: `SubmitExtrinsic`
  --> test/parachain/tests/integration_test.rs:47:6
   |
47 |         fn submit_extrinsic(extrinsic: String) -> PHash;
   |            ^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(dead_code)]` on by default

warning: method is never used: `next_request`
  --> test/parachain/tests/integration_test.rs:45:2
   |
45 |     Author {
   |     ^^^^^^

warning: variant is never constructed: `CurrentBlockHash`
  --> test/parachain/tests/integration_test.rs:52:6
   |
52 |         fn current_block_hash() -> PHash;
   |            ^^^^^^^^^^^^^^^^^^

warning: variant is never constructed: `Header`
  --> test/parachain/tests/integration_test.rs:55:6
   |
55 |         fn header(hash: PHash) -> Option<Header>;
   |            ^^^^^^

warning: variant is never constructed: `BlockHash`
  --> test/parachain/tests/integration_test.rs:58:6
   |
58 |         fn block_hash(hash: Option<u64>) -> Option<PHash>;
   |            ^^^^^^^^^^

warning: method is never used: `next_request`
  --> test/parachain/tests/integration_test.rs:50:2
   |
50 |     Chain {
   |     ^^^^^

warning: method is never used: `current_block_hash`
  --> test/parachain/tests/integration_test.rs:52:3
   |
52 |         fn current_block_hash() -> PHash;
   |         ^^

warning: variant is never constructed: `RuntimeVersion`
  --> test/parachain/tests/integration_test.rs:63:6
   |
63 |         fn runtime_version() -> RuntimeVersion;
   |            ^^^^^^^^^^^^^^^

warning: method is never used: `next_request`
  --> test/parachain/tests/integration_test.rs:61:2
   |
61 |     State {
   |     ^^^^^
```

The code is located [there](https://github.com/paritytech/cumulus/blob/a17dd3464943abf0f36b537a5a23d36d6e8fc7ed/test/parachain/tests/integration_test.rs#L44-L65).

It is strange because everything is used. But I only use it as client, not as server, maybe that is why I get so many warnings.

**Comment by melynx** [comment]
Any follow-up on how to suppress the warnings?

**Comment by cecton** [comment]
Ah no unfortunately I completely forgot about this.

But I supposed you can add `#[allow(unused_variables)]` in front of the macro in your code.

**Comment by niklasad1** [comment]
This is not worked on currently, the crate is being redesigned in the `v2` after some issues but we haven't ported the macros yet.

**Comment by cecton** [comment]
So I guess in the mean time it's best to use the `#[allow(...)]`. I remember it worked for me

**Comment by niklasad1** [comment]
Yes, or `!#[allow(...)]` :)

**Comment by cecton** [comment]
Yeah if you have only that in your file that would be okay. In my original issue I had other things in the file :)

**Comment by cecton** [comment]
Oh! OK I did use the `#![allow(...)]` in my code back then. Bad Cecile. (Maybe it didn't work with the scoped one)

**Comment by niklasad1** [comment]
Fixed by #226
